miranda/io/utils.py

```diff
@@ -69,8 +69,13 @@ def name_output_file(
             raise NotImplementedError(
                 f"Too many `data_vars` in Dataset: {' ,'.join(ds_or_dict.data_vars.keys())}."
             )
-        facets["frequency"] = ds_or_dict.attrs.get("frequency")
-        facets["institution"] = ds_or_dict.attrs.get("institution")
+        for f in ["frequency", "institution", "bias_adjust_project", "domain"]:
+            facets[f] = ds_or_dict.attrs.get(f)
+        if project in ["NEX-GDDP-CMIP6"]:
+            facets["source"] = ds_or_dict.attrs.get("cmip6_source_id")
+            facets["institution"] = ds_or_dict.attrs.get("cmip6_institution_id")
+            facets["member"] = ds_or_dict.attrs.get("variant_label")
+            facets["experiment"] = ds_or_dict.attrs.get("scenario")
```

Comment:
This logic feels like it belongs in the `decode` module. Maybe we can build these configurations into there? I don't like the idea of having lots of special cases in this function.

Reply:
I'm working on building a `decode_nex_gddp_cmip6` based on the existing `decode_cmip6`, but I'm not so sure how to use/where to it ? Could you point me to an example of it being used somewhere else ?